lib/core/services/shared_preference/location_helper.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flowery_e_commerce/di/di.dart';
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:geocoding/geocoding.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import '../../../features/auth/data/data_sources/contracts/offline_data_source.dart';
+class LocationHelper {
+  final OfflineDataSource offlineDataSource = getIt<OfflineDataSource>();
+
+  Future<void> saveLocation(String location) async {
+    try {
+      SharedPreferences prefs = await SharedPreferences.getInstance();
+      String? token = await offlineDataSource.getToken();
```

Comment:
you don't need to concatenate location kye with token

i prefer just save location value with location key